contrib/libbcrypt-cmake/CMakeLists.txt

```diff
@@ -16,4 +16,13 @@ set(SRCS
 
 add_library(_bcrypt ${SRCS})
 target_include_directories(_bcrypt SYSTEM PUBLIC "${LIBRARY_DIR}")
+# Avoid conflicts for crypt_r on FreeBSD [1]:
+#
+# - char *crypt_r(__const char *key, __const char *setting, void *data);
+# - char *crypt_r(const char *, const char *, struct crypt_data *);
+#
+#   [1]: https://github.com/freebsd/freebsd-src/commit/5f521d7ba72145092ea23ff6081d8791ad6c1f9d
+#
+# NOTE: ow-crypt.h is unsed only internally, so PRIVATE is enough
+target_compile_definitions(_bcrypt PRIVATE -D__SKIP_GNU)
```

Comment:
But looks like it will affect all platforms, not only freebsd? What this compile option do?

Reply:
That was the intention to show the problem in all builds if there is any.

As for the option itself, it just removes the declaration for the function - https://github.com/rg3/libbcrypt/blob/8aa32ad94ebe06b76853b0767c910c9fbf7ccef4/crypt_blowfish/ow-crypt.h#L25-L28

Reply:
Can you explain the problem? We have FreeBSD build in CI, it works. All other builds work to.

Reply:
unistd.h from the sysroot does not has definition of crypt_r like freebsd has - https://github.com/freebsd/freebsd-src/commit/5f521d7ba72145092ea23ff6081d8791ad6c1f9d

Reply:
Hm, actually it should use sysroot (i.e. --gcc-toolchain)... and this should not be required

Reply:
No, it should not, it is for cross compiling only